en/flying/terrain_following_holding.md

```diff
@@ -0,0 +1,81 @@
+# Terrain Following/Holding & Range Aid
+
+PX4 supports [Terrain Following](#terrain_following) and [Terrain Hold](#terrain_hold) in [Position](../flight_modes/position_mc.md) and [Altitude modes](../flight_modes/altitude_mc.md), on *multicopters* and *VTOL vehicles in MC mode* that have a [distance sensor](../sensor/rangefinders.md).
+
+PX4 also supports [using a distance sensor as the primary source of altitude data](#distance_sensor_primary_altitude_source) in any mode, either all the time, or just when flying at low altitudes ([Range Aid](#range_aid)).
+
+
+## Terrain Following {#terrain_following}
+
+When *terrain following* is enabled, a vehicle will automatically maintain a relatively constant height above *ground level* if traveling at low altitudes.
+This is useful for avoiding obstacles and for maintaining constant height (e.g. for aerial photography) when flying over varied terrain.
+
+*Terrain following* uses the output of the EKF estimator to provide the altitude estimate, and the estimated terrain altitude (calculated from distance sensor measurements using another estimator) to provide the altitude setpoint.
+As the distance to ground changes, the altitude setpoint adjusts to keep the height above ground constant.
+
+At higher altitudes (when the estimator reports that the distance sensor data is invalid) the vehicle switches to *altitude following*, and will typically fly at a near-constant height above mean sea level (AMSL) using the barometer for altitude data.
+
+> **Note** More precisely, the vehicle will use the *primary source of altitude data* as defined in [EKF2_HGT_MODE](../advanced_config/parameter_reference.md#EKF2_HGT_MODE).
+    This is, by default, the barometer.
+
+Terrain following is enabled by setting [MPC_ALT_MODE](../advanced_config/parameter_reference.md#MPC_ALT_MODE) to `1`.
+
+
+## Terrain Hold {#terrain_hold}
+
+Terrain holding uses a distance sensor to help a vehicle to better maintain a constant height above ground in [position](../flight_modes/position_mc.md) or [altitude](../flight_modes/altitude_mc.md) modes, when horizontally stationary at low altitude.
+This allows a vehicle to avoid altitude changes due to barometer drift or excessive barometer interference from rotor wash.
+
+> **Note** Terrain hold is enabled only in *multicopters* and *VTOL vehicles in MC mode*, and can be used in [position](../flight_modes/position_mc.md) and [altitude modes](../flight_modes/altitude_mc.md).
+
+When moving horizontally (`speed >` [MPC_HOLD_MAX_XY](../advanced_config/parameter_reference.md#MPC_HOLD_MAX_XY), or above the altitude where the distance sensor is providing valid data, the vehicle will instead use the *primary source of altitude data* ([EKF2_HGT_MODE](../advanced_config/parameter_reference.md#EKF2_HGT_MODE)).
```

Comment:
@hamishwillee I think you mean the right thing but it's not formulated correctly. I think it should be:
...or above the altitude where the distance sensor is providing valid data, the vehicle will switch into *altitude following*.
It's important to understand that his has nothing to do with the estimator, it's the position controller switching from terrain hold into altitude following.